.travis.yml

```diff
@@ -16,15 +24,22 @@ before_install:
   - "/sbin/start-stop-daemon --start --quiet --pidfile /tmp/custom_xvfb_99.pid --make-pidfile --background --exec /usr/bin/Xvfb -- :99 -ac -screen 0 1280x1024x16 -extension RANDR"
 
 before_script:
-  - npm install -g jpm mozilla-download
+  - pwd
+  - ls node_modules || true
   - npm install
+  - ls node_modules || true
+  - ls node_modules/.bin || true
+  - ls ~/node_modules || true
+  - npm config ls -l
+  - find / -name node_modules
+  - export JPM_FIREFOX_BINARY=`which firefox`
+  - firefox -v
+  - echo $JPM_FIREFOX_BINARY
   - make all
   - cd ..
-  - mozilla-download --branch mozilla-central --product firefox firefox
-  - export JPM_FIREFOX_BINARY=$TRAVIS_BUILD_DIR/../firefox/firefox
   - cd $TRAVIS_BUILD_DIR/
```

Comment:
According to https://docs.travis-ci.com/user/environment-variables#Default-Environment-Variables

> `TRAVIS_BUILD_DIR`: The absolute path to the directory where the repository being built has been copied on the worker.

Since the `TRAVIS_BUILD_DIR` is absolute, you probably don't need to `cd ..` directly before this. Or maybe don't even need this at all if you're not going up into the parent directory (especially since the first command in the `script` section does a `cd $TRAVIS_BUILD_DIR/build/addon` anyways).